src/test/scala/cognite/spark/v1/FlexibleDataModelCorePropertyRelationTest.scala

```diff
@@ -493,20 +825,36 @@ class FlexibleDataModelCorePropertyRelationTest
       cursor = ""
     )
 
-    val readAllDf = readRows(
+    val readEdgesDfViewAll = readRows(
       instanceType = InstanceType.Edge,
       viewSpaceExternalId = spaceExternalId,
       viewExternalId = viewAll.externalId,
       viewVersion = viewAll.version,
       instanceSpaceExternalId = spaceExternalId
-    ).unionAll(
-      readRows(
-        instanceType = InstanceType.Node,
-        viewSpaceExternalId = spaceExternalId,
-        viewExternalId = viewAll.externalId,
-        viewVersion = viewAll.version,
-        instanceSpaceExternalId = spaceExternalId
-      ))
+    )
+
+    val readNodesDfViewAll = readRows(
+      instanceType = InstanceType.Node,
+      viewSpaceExternalId = spaceExternalId,
+      viewExternalId = viewAll.externalId,
+      viewVersion = viewAll.version,
+      instanceSpaceExternalId = spaceExternalId
+    )
+
+
+    val edgesColumns = readEdgesDfViewAll.columns.toSet
+    val nodesColumns = readNodesDfViewAll.columns.toSet
+
+    val edgesMissingCols = nodesColumns -- edgesColumns
+    val nodesMissingCols = edgesColumns -- nodesColumns
+
+    val nodesWithAllCols = nodesMissingCols.foldLeft(readNodesDf) {
+      (df, col) => df.withColumn(col, lit(null))
+    }
+    val edgesWithAllCols = edgesMissingCols.foldLeft(readEdgesDf) {
+      (df, col) => df.withColumn(col, lit(null))
+    }
+    val readAllDf = nodesWithAllCols.unionAll(edgesWithAllCols)
```

Comment:
obviously this is not ideal... My understanding is that we used to return null columns in type, startNode and endNode when doing this, but obviously this is not possible if you want to also support node "type" property, because then you are unioning the edge.type and the node's view type as the same column.

This test passes if you artificially increase the number of column to match, as done here. Remains to be seen if it won't break something else in the jetfire backend or the sql queries customers might do. I wonder if-or how-joining edges and nodes instances of the same view would make or be possible in sql